dgl_ptm/dgl_ptm/model/data_collection.py

```diff
@@ -62,7 +63,7 @@ def _edge_property_collector(agent_graph, epath, edata, timestep, format, mode):
                                             ))
             for prop in edata:
                 _check_eprop_in_graph(agent_graph, prop)
-                edge_data_instance = edge_data_instance.assign(property=(['n_edges','time'], agent_graph.edata[prop][:,None].cpu().numpy()))
+                edge_data_instance = edge_data_instance.assign(property=(['n_edges','time'], da.from_array(agent_graph.edata[prop][:,None].cpu().numpy())))
```

Comment:
Interesting, so casting the numpy array to a dask array is sufficient to provide the speed up?! nice

Reply:
yeah I was surprised by that too! Also apparently you cannot convert torch tensor to dask array without numpy in the middle.